File: python/src/Creational/Abstract_Factory/BostonConcreteSoupFactory.py

#  BostonConcreteSoupFactory  One of Two concrete factories extending the abstract factory

from AbstractSoupFactory import AbstractSoupFactory
from common.BostonClamChowder import BostonClamChowder
from common.BostonFishChowder import BostonFishChowder

class BostonConcreteSoupFactory(AbstractSoupFactory) :
	def __init__(self) :
		self.factoryLocation = "Boston"

	def makeClamChowder(self) : 
		return BostonClamChowder()

	def makeFishChowder(self) : 
		return BostonFishChowder()

File: ruby/src/Creational/Abstract_Factory/BostonConcreteSoupFactory.rb

#  BostonConcreteSoupFactory  One of Two concrete factories extending the abstract factory

require "Creational/Abstract_Factory/AbstractSoupFactory"
require "common/BostonClamChowder"
require "common/BostonFishChowder"

class BostonConcreteSoupFactory < AbstractSoupFactory
	def initialize()
		@factoryLocation = "Boston"
	end

	def makeClamChowder()
 		return BostonClamChowder.new
	end

	def makeFishChowder()
 		return BostonFishChowder.new
	end

end
